test/helper.rb

```diff
@@ -1,28 +1,13 @@
-require 'rubygems'
-require 'bundler'
+$LOAD_PATH.unshift(File.expand_path("../../", __FILE__))
```

Comment:
Do we need this?

Reply:
I think that we don't need this.

It's for requiring files under `test/` such as `test/xxx.rb` by `require "xxx"`.
I think that we should use `require_relative "xxx"` instead of modifying `$LOAD_PATH` global variable.
And we always use `require_relative "../helper"` in `test/**/test_*.rb`.

Reply:
Thanks!
It seems that we should fix the template too!

There is still something I don't understand.

> It's for requiring files under `test/` such as `test/xxx.rb` by `require "xxx"`.
> I think that we should use `require_relative "xxx"` instead of modifying `$LOAD_PATH` global variable.
> And we always use `require_relative "../helper"` in `test/**/test_*.rb`.

In my local repository: `/test/fluent-plugin-flowcounter-simple`, the paths in `test/helper.rb` are as follows.

* `__FILE__`: `"/test/fluent-plugin-flowcounter-simple/test/helper.rb"`
* `File.expand_path("../../", __FILE__)`: `"/test/fluent-plugin-flowcounter-simple"`

So this doesn't do anything to enable requiring files under `test/`.
(It enables requiring files directly under the repository's root.)

Since we have the following process, we can use `require helper` without using `$LOAD_PATH`.

https://github.com/fluent-plugins-nursery/fluent-plugin-flowcounter-simple/blob/5081e88f97156e019115cab2b4cd810c28ca2ed0/Rakefile#L5-L6

The same is true for the template.

* https://github.com/fluent/fluentd/blob/v1.16.0/templates/new_gem/Rakefile#L7

This is from the beginning, and I can't find the reason from the following.

* https://github.com/fluent/fluentd/issues/1376
* https://github.com/fluent/fluentd/commit/da3664e44ff5974d8f8097094884edd10a8bc41a#diff-3167ac8636d753c8cc2208b3a3dd46770e00d23ab1b4fa483ddd275c840fa41e

Anyways, it doesn't seem necessary for this plugin, so I will remove this.

Reply:
OK. Then it's just a bug of the template.

Reply:
> I can't find the reason why the template fixes `$LOAD_PATH`.

Perhaps this is just for enabling using relative paths from the repository's root such as `require test/xxx`.

So the conclusion is this after all.

> I think that we should use `require_relative "xxx"` instead of modifying `$LOAD_PATH` global variable.

Reply:
> OK. Then it's just a bug of the template.

Yes, we need to consider improving the template.

First, I will remove this from this plugin. Thanks!

Reply:
Fixed at https://github.com/fluent-plugins-nursery/fluent-plugin-flowcounter-simple/compare/55b4cf4093e976a26648c869f4060012e6f9a210..515d2d6fec584370fe39e9a4cebeba68293903b6